BNBRewardApe

DeFiFoFum opened the pull request.

ape-swap commented on migrations/2_deploy_contracts.js:
```diff
@@ -23,56 +31,78 @@ module.exports = async function(deployer, network, accounts) {
+            return deployer.deploy(MasterApe, 
+                BananaToken.address,                        // _banana
+                BananaSplitBar.address,                     // _bananaSplit
+                currentAccount,                             // _devaddr
```
Should we consider making the production the address something different? 

Maybe a multi sig wallet created by all of us.

ape-swap reviewed (approved):
Great work setting up the deployment!

DeFiFoFum committed:
add bnb rewards ape contract, update README

DeFiFoFum committed:
[refactor] Remove need for WBNB
In order to simplify the contract a bit we have removed the need to link a WBNB contract and instead store BNB directly in the contract.

DeFiFoFum committed:
update comment

DeFiFoFum committed:
Saftey checks and improvements
- To integrate this pool seemlessly with the UI, the `bonusEndBlock` variable has be reimplemented.
- Visibilities changed to `external` where allowed
- Saftey checks added so that a reward balance over the current rewards cannot cause a revert.

DeFiFoFum committed:
Merge branch 'master' into feature/bnb-reward-ape

Apegurus reviewed (approved):
Nice work DeFi ape!
